pipeline-marketplace-images/github-sign-off/create-and-wait-for-approval

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+source create-issue
+source wait-approval
+
+# Run create_issue and safely capture its output and exit code
+set +e
+issue=$(create_issue)
+exit_code=$?
+set -e
+
+# 2 = skip-approval or auto-approve => exit cleanly
+if [ "$exit_code" -eq 2 ]; then
+  echo "Skipping approval process as per configuration or auto-approve enabled."
+  exit 0
+fi
+
+# 1 = issue already exists or was auto-approved previously
+# In that case, re-fetch it from the object
+if [ "$exit_code" -eq 1 ]; then
+  issue=$(yq -r '.status.approvalIssue' /kratix/input/object.yaml)
+fi
+
+# Validate the issue string before continuing
+if [ -z "$issue" ] || [ "$issue" == "null" ]; then
+  echo "❌ ERROR: No approval issue found or created. Cannot continue."
+  exit 1
+fi
+
+# Proceed to wait for approval
+wait_for_approval "$issue"
```

Comment:
Will this work? I am not sure how the issue number will be in the status from just this script. Happy to look at a live example if you have one e2e as that is just based on reading the code.